From b7d8d750d7a2efcdccd3af22ad9111c23cebbd8f Mon Sep 17 00:00:00 2001
From: Brli <brli@chakralinux.org>
Date: Sun, 19 Mar 2023 21:48:16 +0800
Subject: [PATCH] remove mImageRegion from nsMenuObject.cpp

---
 widget/gtk/nsMenuObject.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/widget/gtk/nsMenuObject.cpp b/widget/gtk/nsMenuObject.cpp
index 60ff3cb..6860883 100644
--- a/widget/gtk/nsMenuObject.cpp
+++ b/widget/gtk/nsMenuObject.cpp
@@ -197,9 +197,6 @@ nsMenuObjectIconLoader::LoadIcon(const ComputedStyle *aComputedStyle)
         imageRequest = list->mListStyleImage.GetImageRequest();
         if (imageRequest) {
             imageRequest->GetURI(getter_AddRefs(uri));
-            auto& rect = list->mImageRegion.AsRect();
-            imageRect = rect.ToLayoutRect().ToNearestPixels(
-                            pc->AppUnitsPerDevPixel());
         }
     }
 
-- 
2.40.0

