From 911dec1fabd31443147fb386d2ea6605a1ba431b Mon Sep 17 00:00:00 2001
From: BrLi <brli@chakralinux.org>
Date: Mon, 27 Jul 2026 08:17:47 +0800
Subject: [PATCH 2/2] skip creation of user directory extensions

---
 toolkit/xre/nsXREDirProvider.cpp | 21 ++++++++++++++++-----
 1 file changed, 16 insertions(+), 5 deletions(-)

diff --git a/toolkit/xre/nsXREDirProvider.cpp b/toolkit/xre/nsXREDirProvider.cpp
index dc4993ac554..641ead4ca53 100644
--- a/toolkit/xre/nsXREDirProvider.cpp
+++ b/toolkit/xre/nsXREDirProvider.cpp
@@ -1277,6 +1277,13 @@ nsresult nsXREDirProvider::AppendSysUserExtensionPath(nsIFile* aFile) {
   NS_ASSERTION(aFile, "Null pointer!");
 
   nsresult rv;
+  bool exists;
+
+  // check if we are using legacy home
+  const char* legacyhomedir = PR_GetEnv("MOZ_LEGACY_HOME");
+  if (legacyhomedir) {
+    exists = legacyhomedir[0] == '1';
+  }
 
 #if defined(XP_MACOSX) || defined(XP_WIN)
 
@@ -1285,8 +1292,11 @@ nsresult nsXREDirProvider::AppendSysUserExtensionPath(nsIFile* aFile) {
   NS_ENSURE_SUCCESS(rv, rv);
 
   static const char* const sExtensions = "Extensions";
-  rv = aFile->AppendNative(nsDependentCString(sExtensions));
-  NS_ENSURE_SUCCESS(rv, rv);
+
+  if (exists) {
+    rv = aFile->AppendNative(nsDependentCString(sExtensions));
+    NS_ENSURE_SUCCESS(rv, rv);
+  }
 
 #elif defined(XP_UNIX)
 
@@ -1297,9 +1307,10 @@ nsresult nsXREDirProvider::AppendSysUserExtensionPath(nsIFile* aFile) {
 # endif
 
   static const char* const sExtensions = "extensions";
-  rv = aFile->AppendNative(nsDependentCString(sExtensions));
-  NS_ENSURE_SUCCESS(rv, rv);
-
+  if (exists) {
+    rv = aFile->AppendNative(nsDependentCString(sExtensions));
+    NS_ENSURE_SUCCESS(rv, rv);
+  }
 #else
 #  error "Don't know how to get XRE user extension path on your platform"
 #endif
-- 
2.55.0

